tutorials/transparent-proxy-and-filtering-on-k8s-with-initializers/index.md

```diff
@@ -0,0 +1,214 @@
+
+2. Clone the repository containing the code for this tutorial:
+
+        git clone https://github.com/danisla/kubernetes-tproxy
+        cd kubernetes-tproxy
+
+    The remainder of this tutorial will be run from the root of the cloned repository directory.
+
+## Create Container Engine cluster and install Helm
+
+1. Create Container Engine cluster with alpha features enabled, RBAC support, and a cluster version of at least 1.7 to support initializers:
+
+        gcloud container clusters create tproxy-example \
+        --zone us-central1-f \
+        --machine-type n1-standard-1 \
+        --num-nodes 3 \
+        --enable-kubernetes-alpha \
+        --cluster-version 1.7.6 \
+        --no-enable-legacy-authorization
+
+    This command also automatically configures the kubectl command to use the cluster.
+
+2. Create a service account and cluster role binding for Helm to enable RBAC support:
+
+        kubectl create serviceaccount tiller --namespace kube-system
+
+        kubectl create clusterrolebinding tiller-cluster-rule \
+        --clusterrole=cluster-admin \
+        --serviceaccount=kube-system:tiller
+
+3. Install the Helm tool locally in your Cloud Shell instance:
+
+        curl -sL https://storage.googleapis.com/kubernetes-helm/helm-v2.5.1-linux-amd64.tar.gz | tar -xvf - && sudo mv linux-amd64/helm /usr/local/bin/ && rm -Rf linux-amd64
+
+4. Initialize Helm with the service account:
+
+        helm init --service-account=tiller
+
+    This installs the server side component of Helm, Tiller,  in the Kubernetes cluster. The Tiller pod may take a minute to start, run the command below to verify it has been deployed:
+
+        helm version
+
+    You should see the Client and Server versions in the output:
+
+        Client: &version.Version{SemVer:"v2.5.1", GitCommit:"7cf31e8d9a026287041bae077b09165be247ae66", GitTreeState:"clean"}
+        Server: &version.Version{SemVer:"v2.5.1", GitCommit:"7cf31e8d9a026287041bae077b09165be247ae66", GitTreeState:"clean"}
+
+## Install the Helm chart
+
+Before installing the chart, you must first extract the certificates generated by mitmproxy. The generated CA cert is used in the example pods to trust the proxy when making HTTPS requests.
+
+1. Extract the generated certs using Docker:
+
+        cd charts/tproxy
+        docker run --rm -v ${PWD}/certs/:/home/mitmproxy/.mitmproxy mitmproxy/mitmproxy >/dev/null 2>&1
+
+2. Install the chart with the initializer and RBAC enabled:
+
+        helm install -n tproxy --set tproxy.useInitializer=true,tproxy.useRBAC=true .
+
+    The output of this command shows you how to augment your deployments to use the initializer. Example output below:
+
+        Add this metadata annotation to your deployment specs to apply the tproxy initializer:
+
+        metadata:
+        annotations:
+            "initializer.kubernetes.io/tproxy": "true"
+
+3. Run the command below to get the status of the DaemonSet pods:
+
+        kubectl get pods -o wide
```

Comment:
Looks like you got an extra space at the beginning of the command according to GitHub markdown rendering. This will cause bash to not to remember the command.

Reply:
Hmm, I don't see it, can you confirm there is an issue?

Reply:
I can't confirm as I'm not capable of doing what devsite markdown renderer (or whatever we use) does. If I click preview for this file, I see a trailing extra space on GitHub UI that's all I wanted to say.

Reply:
I think it will be ok, my other tutorials are written this way and the final conversion to the website looks OK.